### PR TITLE
Basin UI: Add/Remove Liquidity page

### PR DESCRIPTION
**Pull request by uncoolzero**
Created Add/Remove Liquidity page

### CODE REVIEW

File: projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx
Comment (0xalecks) [inline projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx:1]: In single token mode, you have to click the actual radio... can we make the click target the full row?

Comment (0xalecks) [reply projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx:1]: The UX for Remove multiple in unbalanced proportions is wonky. If you type in a big enough number the quotedetails dissapear. Can we fix this somehow? I'm ok disabling this (making the checkbox for Balanced checked and disabled so you can't change it)

File: projects/dex-ui/src/components/Liquidity/AddLiquidity.tsx
Comment (0xalecks) [inline projects/dex-ui/src/components/Liquidity/AddLiquidity.tsx:1]: We should show quote details even if there are approvals pending

File: projects/dex-ui/src/components/Swap/SwapRoot.tsx [hunk 1]
```diff
@@ -10,15 +10,20 @@ import { useAccount } from "wagmi";
...
 import { Log } from "src/utils/logger";
+import { useParams, useSearchParams } from "react-router-dom";
 import { TransactionToast } from "../TxnToast/TransactionToast";
 
 export const SwapRoot = () => {
   const { address: account } = useAccount();
 
+  const [tokenSwapParams, setTokenSwapParams] = useSearchParams();
+  const token1 = tokenSwapParams.get("token1")
+  const token2 = tokenSwapParams.get("token2")
```
Comment (0xalecks) [inline projects/dex-ui/src/components/Swap/SwapRoot.tsx:21]: Please rename these to `fromToken` and `toToken`

File: projects/dex-ui/src/components/Swap/TokenInput.tsx [hunk 2]
```diff
@@ -139,7 +139,7 @@ export const TokenInput: FC<TokenInput> = ({
 
       {showBalance && (
         <BalanceRow>
-          <Balance>Balance {isBalanceLoading ? <Spinner size={12} /> : balance?.[token.symbol].toHuman()}</Balance>
+          <Balance onClick={balance ? () => updateAmount(balance?.[token.symbol].toHuman()) : undefined}>Balance: {isBalanceLoading ? <Spinner size={12} /> : balance?.[token.symbol].toHuman()}</Balance>
```
Comment (0xalecks) [inline projects/dex-ui/src/components/Swap/TokenInput.tsx:142]: There's already a function for this, `handleClickMax()`, we used to have a "Max" button. Just assign that method here

File: projects/dex-ui/src/pages/Well.tsx [hunk 3]
```diff
@@ -35,10 +37,17 @@ export const Well = () => {
 
   useEffect(() => {
     const run = async () => {
-      if (!well?.tokens) return;
+      if (!well) return
```
Comment (0xalecks) [inline projects/dex-ui/src/pages/Well.tsx:40]: why?

**Review by 0xalecks — CHANGES_REQUESTED** [review]

File: projects/dex-ui/src/assets/images/gear.svg
[the remark this answers is not in the document]
Comment (uncoolzero) [reply projects/dex-ui/src/assets/images/gear.svg:1]: Understood! This should simplify things going forward.

File: projects/dex-ui/src/components/Icons.tsx
[the remark this answers is not in the document]
Comment (uncoolzero) [reply projects/dex-ui/src/components/Icons.tsx:1]: We only use ChevronDown in two components at the moment, we are good

File: projects/dex-ui/src/components/Liquidity/AddLiquidity.tsx [hunk 4]
```diff
@@ -98,49 +130,88 @@ export const AddLiquidity = ({ well, txnCompleteCallback, slippage, slippageSett
...
     [amounts]
   );
 
+  const handleBalancedInputChange = useCallback(
+    (index: number) => (amount: TokenValue) => {
+      if (!prices[index]) {
+        setAmounts({ ...amounts, [index]: amount });
+        return;
+      };
+      const amountInUSD = amount.mul(prices[index]!);
```
[the remark this answers is not in the document]
Comment (uncoolzero) [reply projects/dex-ui/src/components/Liquidity/AddLiquidity.tsx]: Now it hopefully will!

File: projects/dex-ui/src/components/Liquidity/AddLiquidity.tsx [hunk 5]
```diff
@@ -26,13 +30,41 @@ export type AddLiquidityQuote = {
   estimate: TokenValue;
 };
 
-export const AddLiquidity = ({ well, txnCompleteCallback, slippage, slippageSettingsClickHandler }: AddLiquidityProps) => {
+export const AddLiquidity = ({ well, txnCompleteCallback, slippage, slippageSettingsClickHandler, handleSlippageValueChange }: AddLiquidityProps) => {
```
[the remark this answers is not in the document]
Comment (uncoolzero) [reply projects/dex-ui/src/components/Liquidity/AddLiquidity.tsx:32]: This is a bit of a nasty piece of prop drilling, though it seems the slippage component will get an overhaul and we will hopefully no longer be doing this kind of stuff

File: projects/dex-ui/src/components/Icons.tsx [hunk 6]
```diff
@@ -7,9 +7,10 @@ export type SVGProps = {
 };
 
 export const ChevronDown = ({ color = "#000", width, height }: SVGProps) => (
-  <svg xmlns="http://www.w3.org/2000/svg" width={width} height={height} viewBox="0 0 9 6" fill="none">
-    <path fill={color} d="M1.058 0 4.5 3.708 7.942 0 9 1.142 4.5 6 0 1.142 1.058 0Z" />
+  <svg xmlns="http://www.w3.org/2000/svg" width={width} height={height} viewBox="0 0 10 6" fill="none">
+    <path d="M1 1L5 5L9 1" stroke={color} stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
```
[the remark this answers is not in the document]
Comment (uncoolzero) [reply projects/dex-ui/src/components/Icons.tsx]: Oh my, fixed!

File: projects/dex-ui/src/components/Liquidity/SlippagePanel.tsx
[the remark this answers is not in the document]
Comment (uncoolzero) [reply projects/dex-ui/src/components/Liquidity/SlippagePanel.tsx:1]: It seems we are moving the slippage component to a modal, so this should change

File: projects/dex-ui/src/components/Liquidity/QuoteDetails.tsx [hunk 7]
```diff
@@ -94,21 +120,123 @@ const QuoteDetails = ({
...
+      <QuoteDetailLine>
+        <QuoteDetailLabel>USD Value</QuoteDetailLabel>
+        <QuoteDetailValue>{`$${tokenUSDValue.toHuman("0,0.00")}`}</QuoteDetailValue>
       </QuoteDetailLine>
       <QuoteDetailLine>
         <QuoteDetailLabel>Price Impact</QuoteDetailLabel>
-        <QuoteDetailValue>{"1.00%"}</QuoteDetailValue>
-        <GearImage src={infoIcon} alt={"More Info"} />
+        <QuoteDetailValue>{`${priceImpact.toHuman("0,0.00")}%`}</QuoteDetailValue>
+        <Icon src={infoIcon} alt={"More Info"} />
```
[the remark this answers is not in the document]
Comment (uncoolzero) [reply projects/dex-ui/src/components/Liquidity/QuoteDetails.tsx]: Moved this icon to the Icons.tsx file, will move the rest in a future PR

File: projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx
Comment (uncoolzero) [reply projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx:1]: Fixed everything! Input disabling the Balance display in Custom Proportions mode was on purpose, perhaps to indicate to the user that they should be using the bottom input fields, this has been changed, and now clicking the Balance display in Custom Proportions mode will revert it back to Balanced Proportions mode and auto-fill the fields as it normally does.

Please give it a try to see if this behavior is intuitive.

File: projects/dex-ui/src/components/Liquidity/AddLiquidity.tsx
Comment (uncoolzero) [reply projects/dex-ui/src/components/Liquidity/AddLiquidity.tsx:1]: Fixed!

File: projects/dex-ui/src/components/Swap/SwapRoot.tsx (hunk 1 above)
Comment (uncoolzero) [reply projects/dex-ui/src/components/Swap/SwapRoot.tsx]: Fixed!

File: projects/dex-ui/src/components/Liquidity/AddLiquidity.tsx [hunk 8]
```diff
@@ -26,13 +30,41 @@ export type AddLiquidityQuote = {
...
+  const bothAmountsZero = useMemo(() => {
+    if (!well.tokens) {
+      return true;
+    }
+
+    if (well.tokens.length === 0) {
+      return true;
+    }
+
+    const nonZeroValues = Object.values(amounts).filter((amount) => amount.value.gt("0")).length;
```
[the remark this answers is not in the document]
Comment (uncoolzero) [reply projects/dex-ui/src/components/Liquidity/AddLiquidity.tsx]: At the moment the SDK blocks the transaction in the validateAmount step when one token is zero

File: projects/dex-ui/src/components/Liquidity/QuoteDetails.tsx [hunk 9]
```diff
@@ -197,22 +192,19 @@ const QuoteDetails = ({
...
     } else if (oldPrice?.eq(TokenValue.ZERO)) {
-      priceDiff = TokenValue.fromHuman(100, 6);
+      return TokenValue.fromHuman(100, 6);
     } else {
-      if (type === LIQUIDITY_OPERATION_TYPE.REMOVE) {
-        priceDiff = newPrice.div(oldPrice).mul(TokenValue.fromHuman(newPrice.gte(oldPrice) ? 100 : -100, 6));
-      } else {
-        priceDiff = newPrice.div(oldPrice).mul(TokenValue.fromHuman(100, 6)).sub(TokenValue.fromHuman(100, 6)).mul(TokenValue.fromHuman(-1, 6));
-      }
+      priceDiff = oldPrice.sub(newPrice).div(newPrice).mul(TokenValue.fromHuman(100, 6))
```
Comment (0xalecks) [inline projects/dex-ui/src/components/Liquidity/QuoteDetails.tsx:201]: Not clear what's going on here. Why are you multiplying to 100 with six digits?

**Review by 0xalecks — APPROVED** [review]

File: projects/dex-ui/src/components/Liquidity/QuoteDetails.tsx (hunk 9 above)
Comment (uncoolzero) [reply projects/dex-ui/src/components/Liquidity/QuoteDetails.tsx:201]: This is the price impact calculation, we multiply by 100 to get the percentage

**Pull request merged**